crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -205,93 +216,61 @@ impl VerifiableCredential {
         })
     }
 
-    pub fn sign(&self, bearer_did: &BearerDid) -> ResultOld<String> {
-        // default to first VM
-        let key_id = bearer_did.document.verification_method[0].id.clone();
-        let signer = bearer_did.get_signer(BearerDidGetSignerOptions {
-            verification_method_id: Some(key_id.clone()),
-        })?;
-
-        self.sign_with_signer(&key_id, signer)
-    }
-
-    pub fn sign_with_signer(&self, key_id: &str, signer: Arc<dyn Signer>) -> ResultOld<String> {
-        let mut payload = JwtPayload::new();
-        let vc_claim = JwtPayloadVerifiableCredential {
-            context: self.context.clone(),
-            id: Some(self.id.clone()),
-            r#type: self.r#type.clone(),
-            issuer: Some(self.issuer.clone()),
-            issuance_date: Some(self.issuance_date),
-            expiration_date: self.expiration_date,
-            credential_subject: Some(self.credential_subject.clone()),
-        };
-        payload.set_claim("vc", Some(serde_json::to_value(vc_claim)?))?;
-        payload.set_issuer(self.issuer.to_string());
-        payload.set_jwt_id(&self.id);
-        payload.set_subject(&self.credential_subject.id);
-        payload.set_not_before(&self.issuance_date);
-        payload.set_issued_at(&SystemTime::now());
-        if let Some(exp) = &self.expiration_date {
-            payload.set_expires_at(exp)
-        }
-
-        let jose_signer = JoseSigner {
-            kid: key_id.to_string(),
-            signer,
-        };
-
-        let mut header = JwsHeader::new();
-        header.set_token_type("JWT");
-        let vc_jwt = josekit::jwt::encode_with_signer(&payload, &header, &jose_signer)?;
-
-        Ok(vc_jwt)
-    }
-
-    pub fn verify(vc_jwt: &str) -> ResultOld<Self> {
-        // this function currently only supports Ed25519
-        let header = josekit::jwt::decode_header(vc_jwt)?;
+    // this function currently only supports Ed25519
+    pub fn from_vc_jwt(vc_jwt: &str, verify: bool) -> Result<Self> {
```

Comment:
I'm looking through this function to find ways to break it up into smaller pieces. One candidate is the `if verify` block. There's definitely one or two more that I'll find as I read through

Reply:
same as my prior comment, I err on keeping everything together as breaking it up is only boilerplate if it only has one use